src/fabric/src/fabric2_db.erl

```diff
@@ -194,8 +195,40 @@ list_dbs() ->
 
 
 list_dbs(Options) ->
+    Skip = case fabric2_util:get_value(skip, Options, 0) of
+        N when is_integer(N), N >= 0 -> N
+    end,
+    Callback = fun
+        (DbName, {0, Acc}) -> {0, [DbName | Acc]};
+        (_DbName, {S, Acc}) -> {S - 1, Acc}
+    end,
     fabric2_fdb:transactional(fun(Tx) ->
-        fabric2_fdb:list_dbs(Tx, Options)
+        {_, DbNames} = fabric2_fdb:list_dbs(Tx, Callback, {Skip, []}, Options),
+        lists:reverse(DbNames)
+    end).
+
+
+list_dbs(UserFun, UserAcc0, Options) ->
+    Skip = case fabric2_util:get_value(skip, Options, 0) of
+        N when is_integer(N), N >= 0 -> N
+    end,
+    FoldFun = fun
+        (DbName, {0, Acc}) -> maybe_stop(UserFun({row, [{id, DbName}]}, Acc));
```

Comment:
The accumulator return just as Acc here but we expect {Skip, Acc}. Maybe something like would fix it:

```
prototype/fdb-all-dbs-all-docs-qs-params) $ git diff
diff --git a/src/fabric/src/fabric2_db.erl b/src/fabric/src/fabric2_db.erl
index 7ec11732a..501f68bac 100644
--- a/src/fabric/src/fabric2_db.erl
+++ b/src/fabric/src/fabric2_db.erl
@@ -213,13 +213,13 @@ list_dbs(UserFun, UserAcc0, Options) ->
         N when is_integer(N), N >= 0 -> N
     end,
     FoldFun = fun
-        (DbName, {0, Acc}) -> maybe_stop(UserFun({row, [{id, DbName}]}, Acc));
+        (DbName, {0, Acc}) -> {0, maybe_stop(UserFun({row, [{id, DbName}]}, Acc))};
         (_DbName, {S, Acc}) -> {S - 1, Acc}
     end,
     fabric2_fdb:transactional(fun(Tx) ->
         try
             UserAcc1 = maybe_stop(UserFun({meta, []}, UserAcc0)),
-            UserAcc2 = fabric2_fdb:list_dbs(
+            {_SkipFinal, UserAcc2} = fabric2_fdb:list_dbs(
                     Tx,
                     FoldFun,
                     {Skip, UserAcc1},
```

Reply:
I've implemented `skip` differently and this should be fixed for everything.